catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -69,99 +47,105 @@
-
-            ArrayList<Map.Entry<Serializable, Metacard>> entriesToRemove = new ArrayList<>();
-
-            List<Map.Entry<Serializable, Metacard>> registryUpdates = input.getUpdates()
-                    .stream()
-                    .filter(e -> RegistryUtility.isRegistryMetacard(e.getValue()))
-                    .collect(Collectors.toList());
-
-            for (Map.Entry<Serializable, Metacard> entry : registryUpdates) {
-                Metacard updateMetacard = entry.getValue();
-                Metacard existingMetacard = previousMetacardsMap.get(RegistryUtility.getRegistryId(
-                        updateMetacard));
-
-                if (existingMetacard != null) {
-                    updateMetacard.setAttribute(existingMetacard.getAttribute(Metacard.ID));
-                    this.setMetacardExtID(updateMetacard);
-
-                    if (!updateMetacard.getModifiedDate()
-                            .before(existingMetacard.getModifiedDate())) {
-                        for (String transientAttributeKey : RegistryObjectMetacardType.TRANSIENT_ATTRIBUTES) {
-                            Attribute transientAttribute = updateMetacard.getAttribute(
-                                    transientAttributeKey);
-                            if (transientAttribute == null) {
-                                transientAttribute = existingMetacard.getAttribute(
-                                        transientAttributeKey);
-                                if (transientAttribute != null) {
-                                    updateMetacard.setAttribute(transientAttribute);
-                                }
-                            }
-                        }
-                    } else {
-                        entriesToRemove.add(entry);
-                    }
+        Map<String, Metacard> previousMetacardsMap = previousMetacards.stream()
+                .filter(e -> RegistryUtility.isRegistryMetacard(e)
+                        || RegistryUtility.isInternalRegistryMetacard(e))
+                .collect(Collectors.toMap(RegistryUtility::getRegistryId, Function.identity()));
 
-                }
+        ArrayList<Map.Entry<Serializable, Metacard>> entriesToRemove = new ArrayList<>();
+
+        List<Map.Entry<Serializable, Metacard>> registryUpdates = input.getUpdates()
+                .stream()
+                .filter(e -> RegistryUtility.isRegistryMetacard(e.getValue()))
+                .collect(Collectors.toList());
+
+        for (Map.Entry<Serializable, Metacard> entry : registryUpdates) {
+            Metacard updateMetacard = entry.getValue();
+            Metacard existingMetacard = previousMetacardsMap.get(RegistryUtility.getRegistryId(
+                    updateMetacard));
+
+            if (existingMetacard == null) {
+                continue;
             }
 
-            input.getUpdates()
-                    .removeAll(entriesToRemove);
+            if (!updateMetacard.getModifiedDate()
```

Comment:
We know we are a registry metacard at this point and registry metacards should always have a modified date since that is how we keep track of what is the 'newest' data. General in registry we try to keep the groomers from touching our metacards. I will add a check though just to be on the safe side.